microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/integration/cdi/CdiIntegrationTest.java

```diff
@@ -0,0 +1,155 @@
+package org.jboss.eap.qe.microprofile.openapi.integration.cdi;
+
+import static io.restassured.RestAssured.get;
+import static org.hamcrest.Matchers.equalToIgnoringCase;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.eap.qe.microprofile.openapi.OpenApiDeploymentUrlProvider;
+import org.jboss.eap.qe.microprofile.openapi.OpenApiServerConfiguration;
+import org.jboss.eap.qe.microprofile.openapi.apps.routing.router.RouterApplication;
+import org.jboss.eap.qe.microprofile.openapi.apps.routing.router.rest.legacy.Contact;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.ConfigurationException;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientRelatedException;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Test cases for MP OpenAPI and CDI integration, see https://javaee.github.io/tutorial/jaxrs-advanced004.html
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class CdiIntegrationTest {
```

Comment:
@fabiobrz Just a small note: do you prefer writing CDIIntegrationTest or CdiIntegrationTest? What is better in your opinion?

Reply:
Hi @zhantemirov if it is to me, I'd go for how is it now. But please point me to any resource that can help having a clearer idea in case you're referring to some specific naming convention.

Reply:
@fabiobrz you were right keeping CdiIntegrationTest way of writing. Because we can encounter with "HTTP SID or HTTPS ID" situation. So let's keep it that way as it is. Thanks!

Reply:
Thank you too, @zhantemirov. I'll resolve this conversation following your consideration.

Reply:
I prefer CDI over Cdi, I looked at WF and Quarkus code and both `CDI` and `Cdi` are used, Quarkus uses mainly `CDI`.

CDIIntegrationTest looks weird, what about `IntegrationWithCDITest` ?

Reply:
Hi @rsvoboda, thanks for looking at WF for a comparison. Fixing according to your suggestion that also addresses @zhantemirov concerns about naming conventions, IMHO.